test/testdrive/protobuf-multiple-message.td

```diff
@@ -0,0 +1,64 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test that Protobuf imports are handled correctly.
+
+# define the schema
+$ set the-schema
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+message Message1 {
+    bool b = 1;
+}
+
+message Message2 {
+    int64 value = 1;
+}
+
+
+# setup kafka
+$ kafka-create-topic topic=import-csr partitions=1
+
+
+# Compile the protos for kafka to use
+$ file-append path=the.proto
+\${the-schema}
+
+$ protobuf-compile-descriptors inputs=the.proto output=the.pb
+
+
+# Publish the schema
+$ schema-registry-publish subject=testdrive-import-csr-${testdrive.seed}-value schema-type=protobuf
+\${the-schema}
+
+
+
+# Try to ingest the FIRST message, this works
+$ kafka-ingest topic=import-csr format=protobuf descriptor-file=the.pb message=.Message1 confluent-wire-format=true
+{"b": false}}
+
+> CREATE MATERIALIZED SOURCE import_csr FROM
+  KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-import-csr-${testdrive.seed}'
+  FORMAT PROTOBUF USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+
+# Note that only the first message schema works
```

Comment:
@benesch Note that we do not error out when the proto has multiple messages, the `ccsr` crate is opinionated and always chooses the first one: https://github.com/MaterializeInc/materialize/blob/main/src/ccsr/src/client.rs#L120

Reply:
It's specifically this call to `.first()` that I think should change to produce an error if there is more than one message in the set.

Reply:
I think that that should be a separate pr!

Reply:
Oh shoot, failed to actually link you: https://github.com/MaterializeInc/materialize/blob/45042e12f937bcd5f37ff6a04335b02e834c3321/src/sql/src/pure.rs#L831

Reply:
Recording a discussion from Slack: I'd like to ensure that we never have code on main that can properly decode CSR Proto schemas but silently tolerates multiple messages in the same .proto file. Changing that toleration into a hard error message is a breaking change. Whether we do that in a separate PR first or in this PR is totally fine, but want to make sure we don't do it in a separate PR _later_.

Reply:
this will be resolved by the bottom commit when I force push